[2022-02-01 13:45:50,294] {taskinstance.py:1032} INFO - Dependencies all met for <TaskInstance: spark-test.spark_job manual__2022-02-01T13:45:36.490913+00:00 [queued]>
[2022-02-01 13:45:50,303] {taskinstance.py:1032} INFO - Dependencies all met for <TaskInstance: spark-test.spark_job manual__2022-02-01T13:45:36.490913+00:00 [queued]>
[2022-02-01 13:45:50,303] {taskinstance.py:1238} INFO - 
--------------------------------------------------------------------------------
[2022-02-01 13:45:50,304] {taskinstance.py:1239} INFO - Starting attempt 2 of 3
[2022-02-01 13:45:50,304] {taskinstance.py:1240} INFO - 
--------------------------------------------------------------------------------
[2022-02-01 13:45:50,317] {taskinstance.py:1259} INFO - Executing <Task(SparkSubmitOperator): spark_job> on 2022-02-01 13:45:36.490913+00:00
[2022-02-01 13:45:50,323] {standard_task_runner.py:52} INFO - Started process 182 to run task
[2022-02-01 13:45:50,326] {standard_task_runner.py:76} INFO - Running: ['***', 'tasks', 'run', 'spark-test', 'spark_job', 'manual__2022-02-01T13:45:36.490913+00:00', '--job-id', '10', '--raw', '--subdir', 'DAGS_FOLDER/spark-dag.py', '--cfg-path', '/tmp/tmp_uah_s9k', '--error-file', '/tmp/tmprutuxzxp']
[2022-02-01 13:45:50,327] {standard_task_runner.py:77} INFO - Job 10: Subtask spark_job
[2022-02-01 13:45:50,375] {logging_mixin.py:109} INFO - Running <TaskInstance: spark-test.spark_job manual__2022-02-01T13:45:36.490913+00:00 [running]> on host 2809a30b5d43
[2022-02-01 13:45:50,418] {taskinstance.py:1426} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=***@***.com
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=spark-test
AIRFLOW_CTX_TASK_ID=spark_job
AIRFLOW_CTX_EXECUTION_DATE=2022-02-01T13:45:36.490913+00:00
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-02-01T13:45:36.490913+00:00
[2022-02-01 13:45:50,427] {base.py:79} INFO - Using connection to: id: spark_default. Host: spark://spark, Port: 7077, Schema: , Login: , Password: None, extra: {}
[2022-02-01 13:45:50,429] {spark_submit.py:360} INFO - Spark-Submit cmd: spark-submit --master spark://spark:7077 --conf spark.master=spark://spark:7077 --name Spark Hello World --verbose C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\spark-book-count.py C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\book.txt
[2022-02-01 13:45:51,395] {spark_submit.py:514} INFO - Using properties file: null
[2022-02-01 13:45:51,463] {spark_submit.py:514} INFO - WARNING: An illegal reflective access operation has occurred
[2022-02-01 13:45:51,464] {spark_submit.py:514} INFO - WARNING: Illegal reflective access by org.apache.spark.unsafe.Platform (file:/home/***/.local/lib/python3.7/site-packages/pyspark/jars/spark-unsafe_2.12-3.2.1.jar) to constructor java.nio.DirectByteBuffer(long,int)
[2022-02-01 13:45:51,464] {spark_submit.py:514} INFO - WARNING: Please consider reporting this to the maintainers of org.apache.spark.unsafe.Platform
[2022-02-01 13:45:51,465] {spark_submit.py:514} INFO - WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
[2022-02-01 13:45:51,465] {spark_submit.py:514} INFO - WARNING: All illegal access operations will be denied in a future release
[2022-02-01 13:45:51,510] {spark_submit.py:514} INFO - Parsed arguments:
[2022-02-01 13:45:51,510] {spark_submit.py:514} INFO - master                  spark://spark:7077
[2022-02-01 13:45:51,511] {spark_submit.py:514} INFO - deployMode              null
[2022-02-01 13:45:51,511] {spark_submit.py:514} INFO - executorMemory          null
[2022-02-01 13:45:51,512] {spark_submit.py:514} INFO - executorCores           null
[2022-02-01 13:45:51,512] {spark_submit.py:514} INFO - totalExecutorCores      null
[2022-02-01 13:45:51,513] {spark_submit.py:514} INFO - propertiesFile          null
[2022-02-01 13:45:51,514] {spark_submit.py:514} INFO - driverMemory            null
[2022-02-01 13:45:51,514] {spark_submit.py:514} INFO - driverCores             null
[2022-02-01 13:45:51,515] {spark_submit.py:514} INFO - driverExtraClassPath    null
[2022-02-01 13:45:51,516] {spark_submit.py:514} INFO - driverExtraLibraryPath  null
[2022-02-01 13:45:51,516] {spark_submit.py:514} INFO - driverExtraJavaOptions  null
[2022-02-01 13:45:51,517] {spark_submit.py:514} INFO - supervise               false
[2022-02-01 13:45:51,517] {spark_submit.py:514} INFO - queue                   null
[2022-02-01 13:45:51,518] {spark_submit.py:514} INFO - numExecutors            null
[2022-02-01 13:45:51,519] {spark_submit.py:514} INFO - files                   null
[2022-02-01 13:45:51,519] {spark_submit.py:514} INFO - pyFiles                 null
[2022-02-01 13:45:51,519] {spark_submit.py:514} INFO - archives                null
[2022-02-01 13:45:51,520] {spark_submit.py:514} INFO - mainClass               null
[2022-02-01 13:45:51,520] {spark_submit.py:514} INFO - primaryResource         file:/app/C:%5CUsers%5CAyoubKaskasou%5CDesktop%5Ctools%5Cdocker_env%5C***_and_spark%5Cspark-apps%5Cspark-book-count.py
[2022-02-01 13:45:51,520] {spark_submit.py:514} INFO - name                    Spark Hello World
[2022-02-01 13:45:51,521] {spark_submit.py:514} INFO - childArgs               [C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\book.txt]
[2022-02-01 13:45:51,521] {spark_submit.py:514} INFO - jars                    null
[2022-02-01 13:45:51,522] {spark_submit.py:514} INFO - packages                null
[2022-02-01 13:45:51,522] {spark_submit.py:514} INFO - packagesExclusions      null
[2022-02-01 13:45:51,523] {spark_submit.py:514} INFO - repositories            null
[2022-02-01 13:45:51,524] {spark_submit.py:514} INFO - verbose                 true
[2022-02-01 13:45:51,524] {spark_submit.py:514} INFO - 
[2022-02-01 13:45:51,525] {spark_submit.py:514} INFO - Spark properties used, including those specified through
[2022-02-01 13:45:51,525] {spark_submit.py:514} INFO - --conf and those from the properties file null:
[2022-02-01 13:45:51,526] {spark_submit.py:514} INFO - (spark.master,spark://spark:7077)
[2022-02-01 13:45:51,526] {spark_submit.py:514} INFO - 
[2022-02-01 13:45:51,527] {spark_submit.py:514} INFO - 
[2022-02-01 13:45:51,707] {spark_submit.py:514} INFO - Main class:
[2022-02-01 13:45:51,707] {spark_submit.py:514} INFO - org.apache.spark.deploy.PythonRunner
[2022-02-01 13:45:51,708] {spark_submit.py:514} INFO - Arguments:
[2022-02-01 13:45:51,708] {spark_submit.py:514} INFO - file:/app/C:%5CUsers%5CAyoubKaskasou%5CDesktop%5Ctools%5Cdocker_env%5C***_and_spark%5Cspark-apps%5Cspark-book-count.py
[2022-02-01 13:45:51,709] {spark_submit.py:514} INFO - null
[2022-02-01 13:45:51,709] {spark_submit.py:514} INFO - C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\book.txt
[2022-02-01 13:45:51,710] {spark_submit.py:514} INFO - Spark config:
[2022-02-01 13:45:51,710] {spark_submit.py:514} INFO - (spark.master,spark://spark:7077)
[2022-02-01 13:45:51,710] {spark_submit.py:514} INFO - (spark.app.name,Spark Hello World)
[2022-02-01 13:45:51,711] {spark_submit.py:514} INFO - (spark.submit.pyFiles,)
[2022-02-01 13:45:51,711] {spark_submit.py:514} INFO - (spark.submit.deployMode,client)
[2022-02-01 13:45:51,711] {spark_submit.py:514} INFO - Classpath elements:
[2022-02-01 13:45:51,711] {spark_submit.py:514} INFO - 
[2022-02-01 13:45:51,712] {spark_submit.py:514} INFO - 
[2022-02-01 13:45:51,712] {spark_submit.py:514} INFO - 
[2022-02-01 13:45:51,805] {spark_submit.py:514} INFO - python3: can't open file '/app/C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\spark-book-count.py': [Errno 2] No such file or directory
[2022-02-01 13:45:51,818] {spark_submit.py:514} INFO - log4j:WARN No appenders could be found for logger (org.apache.spark.util.ShutdownHookManager).
[2022-02-01 13:45:51,819] {spark_submit.py:514} INFO - log4j:WARN Please initialize the log4j system properly.
[2022-02-01 13:45:51,819] {spark_submit.py:514} INFO - log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
[2022-02-01 13:45:51,864] {taskinstance.py:1700} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1329, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1455, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1511, in _execute_task
    result = execute_callable(context=context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 182, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 446, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --conf spark.master=spark://spark:7077 --name Spark Hello World --verbose C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\spark-book-count.py C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\book.txt. Error code is: 2.
[2022-02-01 13:45:51,870] {taskinstance.py:1277} INFO - Marking task as UP_FOR_RETRY. dag_id=spark-test, task_id=spark_job, execution_date=20220201T134536, start_date=20220201T134550, end_date=20220201T134551
[2022-02-01 13:45:51,881] {standard_task_runner.py:92} ERROR - Failed to execute job 10 for task spark_job
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/task/task_runner/standard_task_runner.py", line 85, in _start_by_fork
    args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/cli/cli_parser.py", line 48, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/cli.py", line 92, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/cli/commands/task_command.py", line 298, in task_run
    _run_task_by_selected_method(args, dag, ti)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/cli/commands/task_command.py", line 107, in _run_task_by_selected_method
    _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/cli/commands/task_command.py", line 184, in _run_raw_task
    error_file=args.error_file,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/session.py", line 70, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1329, in _run_raw_task
    self._execute_task_with_callbacks(context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1455, in _execute_task_with_callbacks
    result = self._execute_task(context, self.task)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1511, in _execute_task
    result = execute_callable(context=context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 182, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 446, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --conf spark.master=spark://spark:7077 --name Spark Hello World --verbose C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\spark-book-count.py C:\Users\AyoubKaskasou\Desktop\tools\docker_env\***_and_spark\spark-apps\book.txt. Error code is: 2.
[2022-02-01 13:45:51,906] {local_task_job.py:154} INFO - Task exited with return code 1
[2022-02-01 13:45:51,944] {local_task_job.py:264} INFO - 0 downstream tasks scheduled from follow-on schedule check
